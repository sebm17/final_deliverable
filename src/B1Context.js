import { createContext, useContext } from "react";

const B1FileInputContext = createContext();
export const B1FileInputProvider = B1FileInputContext.Provider;
export const useB1FileInput = () => useContext(B1FileInputContext);
