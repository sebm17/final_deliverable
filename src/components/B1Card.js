import React, { useCallback, useMemo } from "react";
import { useB1FileInput } from "../B1Context";

export const B1Card = ({ b1File: { id, mpan, mprn, jad, jtc } }) => {
  const [{ selectedInputId }, dispatch] = useB1FileInput();

  const onBinClick = useCallback(() => {
    dispatch({ type: "REMOVE_B1_FILE", id });
  }, [dispatch, id]);

  const onCardClick = useCallback(() => {
    dispatch({ type: "SET_CURRENT_B1_FILE", id });
  }, [dispatch, id]);

  const cardStyles = useMemo(() => {
    if (selectedInputId === id) {
      return "item selectedCard";
    }
    return "item b1Card";
  }, [id, selectedInputId]);

  return (
    <div className={cardStyles} onClick={onCardClick}>
      <div className="content">
        <div className="header">{id}</div>
        <>
          <div>
            <b>MPAN</b>: {mpan}
          </div>
          <div>
            <b>MPRN</b>: {mprn}
          </div>
          <div>
            <b>Job Appointment Date</b>: {jad}
          </div>
          <div>
            <b>Job Type Code</b>: {jtc}
          </div>
        </>
      </div>
      <i
        onClick={onBinClick}
        className="trash alternate outline icon"
        style={{ color: "red", marginTop: "7px" }}
      ></i>
    </div>
  );
};
