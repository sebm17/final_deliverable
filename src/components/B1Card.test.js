import React from "react";
import { render } from "@testing-library/react";
import { B1Card } from "./B1Card";
import { B1FileInputProvider } from "../B1Context";
import { fireEvent } from "@testing-library/react";

test("card renders as expected", () => {
  const result = render(
    <B1FileInputProvider value={[{ selectedInputId: 2 }, jest.fn()]}>
      <B1Card
        b1File={{
          id: "1",
          mpan: "mpan1",
          mprn: "mprn1",
          jad: "jad1",
          jtc: "jtc1"
        }}
      />
    </B1FileInputProvider>
  );

  expect(result).toMatchSnapshot();
});

test("clicking card dispatches selection action", () => {
  const dispatchFn = jest.fn();

  const { getByTestId } = render(
    <B1FileInputProvider value={[{ selectedInputId: 2 }, dispatchFn]}>
      <B1Card
        b1File={{
          id: "1",
          mpan: "mpan1",
          mprn: "mprn1",
          jad: "jad1",
          jtc: "jtc1"
        }}
      />
    </B1FileInputProvider>
  );

  const card = getByTestId("card");
  fireEvent.click(card);

  expect(dispatchFn).toHaveBeenCalledWith({
    type: "SET_CURRENT_B1_FILE",
    id: "1"
  });
});

test("clicking bin dispatches delete action", () => {
  const dispatchFn = jest.fn();

  const { getByTestId } = render(
    <B1FileInputProvider value={[{ selectedInputId: 2 }, dispatchFn]}>
      <B1Card
        b1File={{
          id: "1",
          mpan: "mpan1",
          mprn: "mprn1",
          jad: "jad1",
          jtc: "jtc1"
        }}
      />
    </B1FileInputProvider>
  );

  const bin = getByTestId("bin");
  fireEvent.click(bin);

  expect(dispatchFn).toHaveBeenCalledWith({ type: "REMOVE_B1_FILE", id: "1" });
});
