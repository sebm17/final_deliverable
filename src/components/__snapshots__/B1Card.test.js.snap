// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card renders as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="item b1Card"
        data-testid="card"
      >
        <div
          class="content"
        >
          <div
            class="header"
          >
            1
          </div>
          <div>
            <b>
              mprn
            </b>
            : 
            mprn1
          </div>
          <div>
            <b>
              mpan
            </b>
            : 
            mpan1
          </div>
          <div>
            <b>
              job appointment date
            </b>
            : 
            jad1
          </div>
          <div>
            <b>
              job type code
            </b>
            : 
            jtc1
          </div>
        </div>
        <i
          class="trash alternate outline icon"
          data-testid="bin"
          style="color: red; margin-top: 7px;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="item b1Card"
      data-testid="card"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          1
        </div>
        <div>
          <b>
            mprn
          </b>
          : 
          mprn1
        </div>
        <div>
          <b>
            mpan
          </b>
          : 
          mpan1
        </div>
        <div>
          <b>
            job appointment date
          </b>
          : 
          jad1
        </div>
        <div>
          <b>
            job type code
          </b>
          : 
          jtc1
        </div>
      </div>
      <i
        class="trash alternate outline icon"
        data-testid="bin"
        style="color: red; margin-top: 7px;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
